### PR TITLE
doc: migration-guide: add notes on SSD1306 Kconfig changes

### PR DESCRIPTION
**Pull request by soburi**
Some SSD1306 Kconfig options replaced by dts to make able to configure per device.
Add the description how to migrate it.

### CODE REVIEW

**Comment by soburi** [comment]
@kartben 

About this matter
https://github.com/zephyrproject-rtos/zephyr/pull/62625#issuecomment-1718988108

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@soburi hey this needs a rebase

**Comment by soburi** [comment]
@fabiobaltieri 
Thanks for the notification.
I've updated it.

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/releases/migration-guide-3.5.rst [hunk 1]
```diff
@@ -169,6 +169,23 @@ Required changes
...
+  You can remove ``CONFIG_SSD1306_DEFAULT`` without any other modification.
+  The ``CONFIG_SSD1306_SH1106_COMPATIBLE`` was used to assert that the device
+  is (compatible with) SH1106.
+  This is replaced by dts compatible declaration. For the sh1106 node,
+  change the ``compatible`` designation from :dtcompatible:`solomon,ssd1306fb` to
+  :dtcompatible:`sinowealth,sh1106`.
+
+  The ``CONFIG_SSD1306_REVERSE_MODE`` also replaced by dts.
+  The ``inversion`` property in :dtcompatible:`solomon,ssd1306fb` is equivariant to
+  the removed ``CONFIG_SSD1306_REVERSE_MODE`` option.
```
Comment (kartben) [inline doc/releases/migration-guide-3.5.rst:187]:
```suggestion
* The configuration options for the SSD1306 display driver can now be provided
  via the Devicetree binding :dtcompatible:`solomon,ssd1306fb`. The following
  Kconfig options: ``CONFIG_SSD1306_DEFAULT``,
  ``CONFIG_SSD1306_SH1106_COMPATIBLE``, and ``CONFIG_SSD1306_REVERSE_MODE`` have
  been removed.

  * You can remove ``CONFIG_SSD1306_DEFAULT`` without any other modification.

  * ``CONFIG_SSD1306_SH1106_COMPATIBLE`` was used to assert that the device is
    (compatible with) SH1106. This has been replaced by a dedicated dts
    compatible declaration. You may update an existing sh1106 node to change the
    ``compatible`` designation from :dtcompatible:`solomon,ssd1306fb` to
    :dtcompatible:`sinowealth,sh1106`.
  * ``CONFIG_SSD1306_REVERSE_MODE`` is now set using the ``inversion-on``
    property of the devicetree node.
```

**Review by kartben — CHANGES_REQUESTED** [review]
some grammar/style changes -- sorry for the late review, and thanks for taking the time to document this!

**Comment by soburi** [comment]
@kartben 
Thank you for the correction.
I reflected and pushed it again.

**Review by kartben — APPROVED** [review]
@soburi thank you!

**Review by fabiobaltieri — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**